core/modules/file.py

```diff
@@ -128,6 +131,84 @@ def thumbnailer(fileSkel, existingFiles, params):
     return resList
 
 
+def cloudfunction_thumbnailer(fileSkel, existingFiles, params):
+    if not conf.get("viur.file.thumbnailerURL", False):
+        raise ValueError("viur.file.thumbnailerURL is not set")
+
+    def getsignedurl():
+        if utils.isLocalDevelopmentServer:
+            signedUrl = utils.downloadUrlFor(fileSkel["dlkey"], fileSkel["name"])
+        else:
+            path = f"""{fileSkel["dlkey"]}/source/{file_name}"""
+            if not (blob := bucket.get_blob(path)):
+                logging.warning(f"Blob {path} is missing from cloud storage!")
+                return None
+            authRequest = requests.Request()
+            expiresAt = datetime.now() + timedelta(seconds=60)
+            signing_credentials = compute_engine.IDTokenCredentials(authRequest, "")
+            contentDisposition = "filename=%s" % fileSkel["name"]
+            signedUrl = blob.generate_signed_url(
+                expiresAt,
+                credentials=signing_credentials,
+                response_disposition=contentDisposition,
+                version="v4")
+        return signedUrl
+
+    def make_request():
+        import requests as _requests
+        headers = {'Content-type': 'application/json'}
+        dataStr = base64.b64encode(json.dumps(dataDict).encode("UTF-8"))
+        sig = utils.hmacSign(dataStr)
+        datadump = json.dumps({"dataStr": dataStr.decode('ASCII'), "sign": sig})
+        return _requests.post(conf["viur.file.thumbnailerURL"], data=datadump, headers=headers)
```

Comment:
I think this could be just: 

```suggestion
        dataStr = base64.b64encode(json.dumps(dataDict).encode("UTF-8"))
        sig = utils.hmacSign(dataStr)
        data = {"dataStr": dataStr.decode('ASCII'), "sign": sig}
        return _requests.post(conf["viur.file.thumbnailerURL"], json=data)
```